### ISSUE TITLE
Default SPI library speeds requests are ignored 

### ISSUE THREAD

**Issue by EmbeddedMan** [chipKIT32/chipKIT-core#343]
Test setup: Fubarino Mini, with the following sketch:

```
#include <SPI.h>

const int slaveSelectPin = 30;
char x;
static SPISettings spi_settings = SPISettings(1000000, MSBFIRST, SPI_MODE0);

void setup() {
  Serial.begin(115200);
  while(!Serial)
  {
    ;
  }
  delay(1000);
  pinMode (slaveSelectPin, OUTPUT);
  SPI.begin(); 
}

void loop() {
  digitalWrite(slaveSelectPin,LOW);
  SPI.beginTransaction(spi_settings);
  SPI.transfer(x);
  SPI.endTransaction(); 
  digitalWrite(slaveSelectPin,HIGH); 
  delay(250);
  x++;
}

```

No matter what speed you put in for the first parameter of the SPISettings(), we always get  a SPI clock rate of 1.538 MHz.

**Comment by EmbeddedMan** [comment]
@majenkotech I know you've worked with our SPI library a bit before - any idea why this might be? I'm trying to dig into it right now to see what I can find out as well.

**Comment by EmbeddedMan** [comment]
I found the problem and I  believe fixed it in pull request #343.

**Comment by EmbeddedMan** [comment]
Pull request merged.
